include/nbl/builtin/hlsl/emulated/float64_t.hlsl

```diff
@@ -395,6 +395,46 @@ namespace hlsl
             return bit_cast<this_t>(data ^ ieee754::traits<float64_t>::signMask);
         }
 
+        static this_t sqrt(this_t number)
+        {
+            // so it doesn't return NaN for -0.0
+            bool isZero = !(number.data & 0x7FFFFFFFFFFFFFFFull);
+            if (isZero)
+                return number;
+
+            bool isNegative = (number.data >> 63) > 0;
+            if (isNegative)
+                return bit_cast<this_t>(ieee754::traits<this_t>::quietNaN);
```

Comment:
note that floats go from 0 to MAX, then INF,, then NaN then signed 0 , then -INF, then NaN

you can test if the `number` is larger than max, then check for INF and signed 0 (if zero or inf, return number again) then return NaN for everything else

Reply:
you can skip caring about `FastMath` and just make fast math behave correctly in this one case